src/MongoDB.Driver.Core/Core/Operations/QueryHelper.cs

```diff
@@ -60,13 +60,28 @@ public static BsonDocument CreateReadPreferenceDocument(ServerType serverType, R
             }
 
             // simple ReadPreferences of Primary and SecondaryPreferred are encoded in the slaveOk bit
-            if (readPreference.ReadPreferenceMode == ReadPreferenceMode.Primary || readPreference.ReadPreferenceMode == ReadPreferenceMode.SecondaryPreferred)
+            //if (readPreference.ReadPreferenceMode == ReadPreferenceMode.Primary || readPreference.ReadPreferenceMode == ReadPreferenceMode.SecondaryPreferred)
+            //{
+            //    var hasTagSets = readPreference.TagSets != null && readPreference.TagSets.Count > 0;
+            //    if (!hasTagSets && !readPreference.MaxStaleness.HasValue && readPreference.Hedge == null)
+            //    {
+            //        return null;
+            //    }
+            //}
```

Comment:
Code below looks more like the prose in the test but should have the exact same effect as lines 63-70

Reply:
I don't care whether we keep lines 63-70 or replace them with the new lines 71-84 (at the risk of introducing a bug?).

Reply:
@jyemin should we keep the original code or switch to the refactored code that looks more like the prose in the spec?

Reply:
I don't care much either.  Up to you, Robert.